assembly/error/index.ts

```diff
@@ -36,7 +36,14 @@ export class Result<T> {
     /** Used by the underlying lunatic call to identify an error if it exists. */
     private errId: u64 = u64.MAX_VALUE,

```

Comment:
Can this be merged with the `value` field? That is, the leading bit can store whether the field is an error or a successful value. A rough example would be:
```ts
private internalValue: u64;

get ok(): bool {
  return !(this.internalValue & (1 << 63));
}

get value(): T {
  assert(this.ok);
  // The leading bit will definitely be zero.
  // This code should work as long as memory addresses are at most 63-bits long.
  return changetype<T>(this.internalValue);
}

private get errId(): u64 {
  assert(!this.ok);
  return this.internalValue & i64.MAX_VALUE;
}
```

Reply:
No because if `T` is managed it messes with the way that the gc handles the reference. `T` is almost always a reference.

Reply:
Can't you manually implement `__visit()`?

Reply:
Maybe. Is it really worth it to fight over four bytes of a reference?